### PR TITLE
Fixed issue with a missing key being returned from the win_system.workgroup state.

### LINKED ISSUE #57790: [BUG] win_system.workgroup with missing "name" key in state return (v3001)
**Description**
Trying to use the `workgroup` state from `salt.states.win_system` results in a missing "name" key and leaves the following droppings in the log.

```
2020-06-25 00:53:54,081 [salt.utils.decorators.state:39  ][DEBUG   ][3532] An exception occurred in this state: The following keys were not present in the state return: name.
Traceback (most recent call last):
  File "c:\python37\lib\site-packages\salt\utils\decorators\state.py", line 34, in _func
    result = pls(result)
  File "c:\python37\lib\site-packages\salt\utils\decorators\state.py", line 78, in content_check
    raise SaltException(err_msg)
salt.exceptions.SaltException: The following keys were not present in the state return: name.
Traceback (most recent call last):
  File "c:\python37\lib\site-packages\salt\utils\decorators\state.py", line 34, in _func
    result = pls(result)
  File "c:\python37\lib\site-packages\salt\utils\decorators\state.py", line 78, in content_check
    raise SaltException(err_msg)
salt.exceptions.SaltException: The following keys were not present in the state return: name.
2020-06-25 00:53:54,081 [salt.minion      :1949][WARNING ][3532] The minion function caused an exception
Traceback (most recent call last):
  File "c:\python37\lib\site-packages\salt\minion.py", line 1870, in _thread_return
    opts, data, func, args, kwargs
  File "c:\python37\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "c:\python37\lib\site-packages\salt\modules\state.py", line 1129, in highstate
    orchestration_jid=orchestration_jid,
  File "c:\python37\lib\site-packages\salt\state.py", line 4468, in call_highstate
    return self.state.call_high(high, orchestration_jid)
  File "c:\python37\lib\site-packages\salt\state.py", line 3209, in call_high
    ret = self.call_chunks(chunks)
  File "c:\python37\lib\site-packages\salt\state.py", line 2481, in call_chunks
    running = self.call_chunk(low, running, chunks)
  File "c:\python37\lib\site-packages\salt\state.py", line 3084, in call_chunk
    self.event(running[tag], len(chunks), fire_event=low.get("fire_event"))
  File "c:\python37\lib\site-packages\salt\state.py", line 2828, in event
    six.text_type(chunk_ret["__run_num__"]),
KeyError: '__run_num__'
2020-06-25 00:53:54,081 [salt.utils.event :319 ][DEBUG   ][3532] SaltEvent PUB socket URI: 4510
2020-06-25 00:53:54,081 [salt.utils.event :320 ][DEBUG   ][3532] SaltEvent PULL socket URI: 4511
2020-06-25 00:53:54,081 [salt.transport.ipc:40  ][TRACE   ][3532] IPCClient: Connecting to socket: 4511
```

**Setup**
Use a state like:
```
blah:
 system.workgroup:
  -name: whatever
```

**Steps to Reproduce the behavior**
Apply it to a minion.

**Expected behavior**
It should apply the workgroup, and continue executing states if necessary.

**Screenshots**

**Versions Report**
```
Salt Version:
           Salt: 3001
 
Dependency Versions:
           cffi: 1.12.2
       cherrypy: 17.4.1
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 3.9.8
   pycryptodome: 3.9.7
         pygit2: Not Installed
         Python: 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 19:29:22) [MSC v.1916 32 bit (Intel)]
   python-gnupg: 0.4.4
         PyYAML: 5.1.2
          PyZMQ: 18.0.1
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: cp1252
        machine: x86
        release: 8.1
         system: Windows
        version: 8.1 6.3.9600 SP0 
```

### PR DESCRIPTION
**Pull request by arizvisa**
# What does this PR do?
The `win_system.workgroup` state was returning a dictionary that did not include the "name" key which resulted in a windows-minion logging an error during the application of a state. This is due to the function re-initializing its return dictionary with an empty one thus clearing out the necessary dictionary item. In some cases this could interfere with a high-state being applied.

### What issues does this PR fix or reference?
fixes #57790 

### Previous Behavior
Previously when calling the `win_system.workgroup` state the dictionary that it would return would be missing the "name" key that was passed as its parameters. During the validation of the return dictionary, an error would be logged and interrupt the state that is currently being applied.

### New Behavior
The line that reassigns the return dictionary has been removed which allows the previous assignment of the "name" key to be returned passing the checks imposed by the content checker decorator.

### Merge requirements satisfied?
- [ ] Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Review by waynew — DISMISSED** [review]
Hey @arizvisa thanks for the patch - will you be writing tests for this? We'll need to have tests before we can merge.

**Comment by arizvisa** [comment]
@waynew, the conversation in issue #55790 said:

```
garethgreenaway commented 23 days ago
@arizvisa Thanks for the report and good find, this looks like the likely culprit. Would you be able to put in a PR with the fix? We can offer assistance on a test if needed.

@arizvisa
Contributor Author
arizvisa commented 23 days ago

Yes. Please. I really don't have time to be writing code and thus have to do this stuff on my off-time (it's 2:30a here), so I'm always hesitant to submit PRs because of the extra need to "maintain" them.

I'll submit it in a second.
```

**Comment by arizvisa** [comment]
I don't do software engineering as a career, nor IT, or even anything remotely-related to devops as we don't even use saltstack at work. All my time spent troubleshooting regular bugs and writing patches for this project is actually contrary of what my job entails since none of these fixes have any kind of security impact. This is why I end up spending non-work (see family) time at super crazy hours trying to keep this thing working. :-/

One of my earlier PRs which refactored the etcd returner has even been locked in porting-hell for almost 3 years due to being developed against "develop" during salt's transition to "master". Now the etcd v3 api is becoming a thing, and that PR still hasn't made it into widespread use yet. Even a port to master was offered and implemented. yet ignored as the "project" is still ongoing. By the time the PR comes out, the etcd v2 api will likely become completely deprecated by then, (not to mention that it'll appear that someone else wrote it in the commit logs due to "porting"..but whatev).

So unfortunately no, there isn't a testcase for this single-line removal. I really hope you can understand why. :( At some point something had changed in the codebase that caused this 3 year old feature to manifest and the committer of that change didn't bother to check it against all of the available modules to see what it would break. For how much code their commits interact with, its testcase should've been significantly more than just a smoke check.

**Comment by garethgreenaway** [comment]
re-run all

**Review by Ch3LL — DISMISSED** [review]

**Review by twangboy — APPROVED** [review]

**Comment by arizvisa** [comment]
ur the man, gareth.

**Pull request merged**
